### PR TITLE
Fix the problem with total spend on statistic page

### LINKED ISSUE #1019: Problem with total spend
There is a problem with to total displayed in statistics.
If i understood correctly : 
    - If Alice buy something 20 for Alice and Bob
    - Bob give her 10 for refund.
In theory, the total spend is 20 but ihatemoney will display 30 because the reimbursement is considered as an expense. The total will always be 1.5 * real value.
We should be able to "tag" the reimbursement in order not to take it into account in the total.

### PR DESCRIPTION
**Pull request by GingLuo**
We're the same team that commented. We were gonna fix this.
Fix the problem with total spend on the statistic page
Fix #1019

### CODE REVIEW

File: ihatemoney/migrations/versions/2dcb0c0048dc_autologger.py [hunk 1]
```diff
@@ -28,6 +28,7 @@ def upgrade():
             "external_link", sa.UnicodeText(), autoincrement=False, nullable=True
         ),
         sa.Column("archive", sa.Integer(), autoincrement=False, nullable=True),
+        sa.Column("is_reimbursement", sa.Boolean(), nullable=True),
```
Comment (Glandos) [inline ihatemoney/migrations/versions/2dcb0c0048dc_autologger.py:31]: This file is auto-generated, please read https://ihatemoney.readthedocs.io/en/latest/contributing.html#creating-database-migrations

File: ihatemoney/migrations/versions/b9a10d5d63ce_.py [hunk 2]
```diff
@@ -49,6 +49,7 @@ def upgrade():
         sa.Column("date", sa.Date(), nullable=True),
         sa.Column("what", sa.UnicodeText(), nullable=True),
         sa.Column("archive", sa.Integer(), nullable=True),
+        sa.Column("is_reimbursement", sa.Boolean(), nullable=True),
```
Comment (Glandos) [inline ihatemoney/migrations/versions/b9a10d5d63ce_.py:52]: This file is auto-generated, please read https://ihatemoney.readthedocs.io/en/latest/contributing.html#creating-database-migrations

File: ihatemoney/models.py [hunk 3]
```diff
@@ -158,7 +158,9 @@ def monthly_stats(self):
         """
         monthly = defaultdict(lambda: defaultdict(float))
         for bill in self.get_bills_unordered().all():
-            monthly[bill.date.year][bill.date.month] += bill.converted_amount
+            if (bill.is_reimbursement == False):
+            # if (not bill.what.startswith("R-")):
+                monthly[bill.date.year][bill.date.month] += bill.converted_amount
```
Comment (Glandos) [inline ihatemoney/models.py:163]:
```suggestion
            if (not bill.is_reimbursement):
                monthly[bill.date.year][bill.date.month] += bill.converted_amount
```

Remove the comment, and don't compare boolean

File: ihatemoney/forms.py [hunk 4]
```diff
@@ -327,6 +327,7 @@ class BillForm(FlaskForm):
     what = StringField(_("What?"), validators=[DataRequired()])
     payer = SelectField(_("Who paid?"), validators=[DataRequired()], coerce=int)
     amount = CalculatorStringField(_("How much?"), validators=[DataRequired()])
+    is_reimbursement = BooleanField("Reimbursement", default=False)
```
Comment (Glandos) [inline ihatemoney/forms.py:330]: Add internationalisation call.
```suggestion
    is_reimbursement = BooleanField(_("Reimbursement"), default=False)
```

File: ihatemoney/templates/settle_bills.html [hunk 5]
```diff
@@ -16,6 +16,11 @@
             <td>{{ bill.ower }}</td>
             <td>{{ bill.receiver }}</td>
             <td>{{ bill.amount|currency }}</td>
+            <!-- <td>
+                <a href="{{ url_for('.settle_paid') }}" class="btn btn-primary " data-toggle="modal" data-keyboard="false" data-target="#bill-form">
+                    Settle
+                </a>
+            </td> -->
```
Comment (Glandos) [inline ihatemoney/templates/settle_bills.html:23]: Don't commit commented code

File: ihatemoney/tests/api_test.py [hunk 6]
```diff
@@ -393,7 +393,8 @@ def test_bills(self):
             "converted_amount": 25.0,
             "original_currency": "XXX",
             "external_link": "https://raclette.fr",
-        }
+            "is_reimbursement": "false",
+	}
```
Comment (Glandos) [inline ihatemoney/tests/api_test.py:397]: Forget about formatting manually with https://ihatemoney.readthedocs.io/en/latest/contributing.html#formatting-code

File: ihatemoney/tests/api_test.py [hunk 7]
```diff
@@ -464,6 +465,7 @@ def test_bills(self):
             "converted_amount": 25.0,
             "original_currency": "XXX",
             "id": 1,
+	    "is_reimbursement": false,
```
Comment (Glandos) [inline ihatemoney/tests/api_test.py:468]: Forget about formatting manually with [ihatemoney.readthedocs.io/en/latest/contributing.html#formatting-code](https://ihatemoney.readthedocs.io/en/latest/contributing.html#formatting-code)

File: ihatemoney/web.py [hunk 8]
```diff
@@ -749,6 +749,28 @@ def add_bill():
...
+            flash(_("The bill has been added"))
+
+            args = {}
+            if form.submit2.data:
+                args["add_bill"] = True
+
+            return redirect(url_for(".list_bills", **args))
+
+    return render_template("add_bill.html", form=form)
+
```
Comment (Glandos) [inline ihatemoney/web.py:773]: I don't see any changes linked to this new route. Was it linked to the commented HTML code?

**Review by Glandos — CHANGES_REQUESTED** [review]
Thanks for this, it's not that bad for a start!

I left some comment and suggestion.
But to be accepted, we need for unit test. Since the goal of this is to avoid adding refund as expenses, a good test would be to check that refunds are not added to the total expense amount.

**Comment by Glandos** [comment]
By the way, there is another implementation in #1116

**Comment by GingLuo** [comment]
Hi Glandos,
Really appreciate the suggestions.
I've seen your comments and #1116, so I wonder should we just keep track of the reimbursement as an enum of bill type? 
For me, I feel like Transfer and Reimbursement are kind of the same thing (i.e. from one person to the other person). So to differentiate between Expense and Reimbursement, adding a new boolean column 'is_reimbursement' in the database also works. Or it might be better to keep the bill type as Expense and Reimbursement.
But their PR is more complete than ours, and we are ok to follow their design.
Let me know if you have any questions.
Thanks

**Comment by zorun** [comment]
Closing because #1290 has been merged

**Pull request closed without merging**